tests/test_connection.py

```diff
@@ -128,6 +128,35 @@ def test_bad_opening(node_factory):
     l2.daemon.wait_for_log('to_self_delay 100 larger than 99')
 
 
+def test_opening_tiny_channel(node_factory):
+    # Try to establish channels with smaller than 1000000 msat (default)
+    #
+    # [l1] ---> [l2]
+    #
+    # BOLT2
+    # The receiving node MAY fail the channel if:
+    #  - funding_satoshis is too small
+    #
+    # Expects: opening is rejected
+    l1 = node_factory.get_node()
+    l2 = node_factory.get_node()
+
+    l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
+
+    min_capacity_sat = 1000
+    dustlimit_sat = 546
+    reserves = 2 * dustlimit_sat
+
+    # Open channel with less than 1000 sats (default) should be rejected
+    with pytest.raises(RpcError, match=r'channel capacity is 999sat, which is below 1000000msat'):
+        l1.fund_channel(l2, min_capacity_sat + reserves - 1)
+
+    # When the capacity is precisely met for the fundee
+    # NOTE: The next line will (currently) raise an exception
```

Comment:
what needs to change to make this pass?

Reply:
typically if you've got a test that you want to make go from 'not passing' to 'passing' in a commit set, you can annotate the test with `@pytest.mark.xfail(strict=True)` e.g. https://github.com/ElementsProject/lightning/pull/2490/commits/a4ec45f5d6a5e43ba5dc2d5d47c45f54dc27657d#diff-af265d267530ebfb09faeaf242e16936R1036

Reply:
@niftynei Well, I'm not sure, as you can see in the changeset of the complete PR #2506 , the actual minimum funding amount for the funder is around 6000sat. Smaller channels will run into these funder side caused related exception.

If someone with deeper knowledge can tell me why, I can document the 'testcase' in a better helpful way.

Reply:
i'm pretty sure this is due to not providing enough to cover the funding tx fee. see https://github.com/ElementsProject/lightning/blob/master/common/initial_commit_tx.c#L62 and also

Reply:
@niftynei Thanks, I will resolve that now in the other PR #2506 so I do not have two diverging test cases... I will also close this PR as the memory leak is also included in the other...